lifetimes/utils.py

```diff
@@ -101,31 +98,93 @@ def summary_data_from_transaction_data(transactions, customer_id_col, datetime_c
 
     transactions = transactions[select_columns].copy()
 
-    def to_period(d):
-        return d.to_period(freq)
+    # make sure the date column uses datetime objects, and use Pandas' DateTimeIndex.to_period()
+    # to convert the column to a PeriodIndex which is useful for time-wise grouping and truncating
+    transactions[datetime_col] = pd.to_datetime(transactions[datetime_col], format=datetime_format)
+    transactions = transactions.set_index(datetime_col).to_period(freq)
+
+    transactions = transactions.ix[(transactions.index <= observation_period_end)].reset_index()
 
-    transactions[datetime_col] = pd.to_datetime(transactions[datetime_col], format=datetime_format).map(to_period)
-    observation_period_end = to_period(pd.to_datetime(observation_period_end, format=datetime_format))
+    if monetary_value_col:
+        # when we have a monetary column, make sure to sum together any values in the same period
+        period_transactions = transactions.groupby([customer_id_col, datetime_col], sort=False, as_index=False).sum()
+    else:
+        period_transactions = reduce_events_to_period(transactions, *select_columns).reset_index()
 
-    transactions = transactions.ix[transactions[datetime_col] <= observation_period_end]
+    # find all of the initial transactions for every customer
+    first_transactions = period_transactions.groupby(customer_id_col, sort=True, as_index=False).first()
+    first_transactions['first'] = True
 
-    # reduce all events per customer during the period to a single event:
-    period_transactions = reduce_events_to_period(transactions, *select_columns).reset_index(level=select_columns[1:])
+    # join the first_transaction column to the transaction log
+    period_transactions = pd.merge(
+        period_transactions,
+        first_transactions[[customer_id_col, datetime_col, 'first']],
+        how="left",
+        on=[customer_id_col, datetime_col]
+    )
+    # boolean invert the 'first' column to give a column representing repeated transactions
+    period_transactions['repeated'] = ~(period_transactions['first'].fillna(False))
```

Comment:
I'm not crazy about how this is computed, that is, it doesn't feel very clear to me. What about something like:

```
def _set_repeated_in_df(df):
    df['repeat'] = True
    df['repeat'].iloc[0] = False
    return df

period_transactions.sort_values(datetime_col).groupby(customer_id_col).apply(_set_repeated_in_df)
```